superset/db_engine_specs/presto.py

```diff
@@ -911,10 +912,14 @@ def where_latest_partition(  # pylint: disable=too-many-arguments
         if values is None:
             return None
 
-        column_names = {column.get("name") for column in columns or []}
+        column_names = {column.get("name") : column.get('type') for column in columns or []}
```

Comment:
Maybe this should be `column_type_by_name` and thus line 918 will be easier to grok. In that case you could roll lines 918 and 919 into one.